### ISSUE TITLE
Otel-contrib-collector arm64 is not getting installed on AWS graviton instances 

### ISSUE THREAD

**Issue by mjindalmotiv8** [open-telemetry/opentelemetry-collector-contrib#6508]
**Describe the bug**
Otel-contrib-collector arm64 is not getting installed on AWS graviton instances and is returning the error intended for a different architecture.

**Steps to reproduce**
1. Launch an EC2 instance using 64 bit ARM  AMI. For my test, I had used the following `Amazon Linux 2 AMI (HVM) - Kernel 5.10, SSD Volume Type - ami-00f7e5c52c0f43726 (64-bit x86) / ami-0c8ae7b6508eff3f4 (64-bit Arm)`
2. Then try to install otel-contrib-collector 64 bit ARM on it
```
[root@ip-10-0-x-x bin]# curl -L https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/download/v0.36.0/otel-contrib-collector-0.36.0-1.arm64.rpm -o otel-contrib-collector-0.36.0-1.arm64.rpm
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   680  100   680    0     0   3269      0 --:--:-- --:--:-- --:--:--  3269
100 47.2M  100 47.2M    0     0  25.0M      0  0:00:01  0:00:01 --:--:-- 31.9M
[root@ip-10-0-x-x bin]#  rpm -Uvh  ./otel-contrib-collector-0.36.0-1.arm64.rpm 
Preparing...                          ################################# [100%]
	package otel-contrib-collector-0.36.0-1.arm64 is intended for a different architecture
[root@ip-10-0-x-x bin]# uname -r
5.10.75-79.358.amzn2.aarch64
```

**What did you expect to see?**
Success on install of otel-contrib-collector-0.36.0-1.arm64.rpm

**What did you see instead?**
Error `package otel-contrib-collector-0.36.0-1.arm64 is intended for a different architecture`

**What version did you use?**
`otel-contrib-collector-0.36.0-1.arm64.rpm`

**What config did you use?**
N/A

**Environment**
OS: Amazon Linux 2 AMI (HVM) - Kernel 5.10
Compiler(if manually compiled): AWS

**Additional context**

**Comment by jpkrohling** [comment]
Questions:

- Is the same happening with the non-RPM version of it? https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/download/v0.40.0/otelcontribcol_linux_arm64
- Is the same happening with the core distribution? https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v0.40.0/otelcol_0.40.0_linux_arm64.rpm
- Are you able to run it if you compile it yourself using `make otelcontribcol-linux_arm64`?

**Comment by mjindalmotiv8** [comment]
- Is the same happening with the non-RPM version of it? https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/download/v0.40.0/otelcontribcol_linux_arm64
The binary runs fine.

- Is the same happening with the core distribution? https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v0.40.0/otelcol_0.40.0_linux_arm64.rpm 
I do not see a rom install error when I use otelcol_0.40.0_linux_arm64.rpm but the service fails to start up. The error I see is
```
Dec  3 16:24:46 ip-10-0-28-210 systemd: Reloading.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Failed to load environment files: No such file or directory
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service failed to run 'start' task: No such file or directory
Dec  3 16:24:46 ip-10-0-28-210 systemd: Failed to start OpenTelemetry Collector.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Unit otelcol.service entered failed state.
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service failed.
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service holdoff time over, scheduling restart.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Stopped OpenTelemetry Collector.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Failed to load environment files: No such file or directory
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service failed to run 'start' task: No such file or directory
Dec  3 16:24:46 ip-10-0-28-210 systemd: Failed to start OpenTelemetry Collector.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Unit otelcol.service entered failed state.
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service failed.
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service holdoff time over, scheduling restart.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Stopped OpenTelemetry Collector.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Failed to load environment files: No such file or directory
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service failed to run 'start' task: No such file or directory
Dec  3 16:24:46 ip-10-0-28-210 systemd: Failed to start OpenTelemetry Collector.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Unit otelcol.service entered failed state.
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service failed.
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service holdoff time over, scheduling restart.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Stopped OpenTelemetry Collector.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Failed to load environment files: No such file or directory
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service failed to run 'start' task: No such file or directory
Dec  3 16:24:46 ip-10-0-28-210 systemd: Failed to start OpenTelemetry Collector.
Dec  3 16:24:46 ip-10-0-28-210 systemd: Unit otelcol.service entered failed state.
Dec  3 16:24:46 ip-10-0-28-210 systemd: otelcol.service failed.
Dec  3 16:24:47 ip-10-0-28-210 systemd: otelcol.service holdoff time over, scheduling restart.
Dec  3 16:24:47 ip-10-0-28-210 systemd: Stopped OpenTelemetry Collector.
Dec  3 16:24:47 ip-10-0-28-210 systemd: Failed to load environment files: No such file or directory
Dec  3 16:24:47 ip-10-0-28-210 systemd: otelcol.service failed to run 'start' task: No such file or directory
Dec  3 16:24:47 ip-10-0-28-210 systemd: Failed to start OpenTelemetry Collector.
Dec  3 16:24:47 ip-10-0-28-210 systemd: Unit otelcol.service entered failed state.
Dec  3 16:24:47 ip-10-0-28-210 systemd: otelcol.service failed.
Dec  3 16:24:47 ip-10-0-28-210 systemd: otelcol.service holdoff time over, scheduling restart.
Dec  3 16:24:47 ip-10-0-28-210 systemd: Stopped OpenTelemetry Collector.
Dec  3 16:24:47 ip-10-0-28-210 systemd: start request repeated too quickly for otelcol.service
Dec  3 16:24:47 ip-10-0-28-210 systemd: Failed to start OpenTelemetry Collector.
Dec  3 16:24:47 ip-10-0-28-210 systemd: Unit otelcol.service entered failed state.
Dec  3 16:24:47 ip-10-0-28-210 systemd: otelcol.service failed.
```

**Comment by jpkrohling** [comment]
Thanks, I sent in a PR to fix the problem with the otelcol core service not starting properly. It should be fixed for 0.41.0. That still won't solve the problem with the packaged arm64 problem you reported originally though. 

It's strange that the binary you downloaded from the releases page works fine, while the binary within the package does not. I'll try to figure this out, but let me know if you have any other hints.

**Comment by jpkrohling** [comment]
> I'll try to figure this out, but let me know if you have any other hints.

I guess I'll have to defer to someone with access to that instance type. @alolita, is this someone on your team could help debug and fix?
